tests/models/decoder_only/language/test_jamba.py

```diff
@@ -65,6 +70,99 @@ def test_batching(
     )
 
 
+@pytest.mark.parametrize("model", MODELS)
+@pytest.mark.parametrize("dtype", ["float16"])
+@pytest.mark.parametrize("max_tokens", [10])
+def test_mamba_prefill_chunking_with_parallel_sampling(
+        hf_runner, vllm_runner, example_prompts, model: str, dtype: str,
+        max_tokens: int) -> None:
+    # Tests prefill chunking in conjunction with n>1, in this case,
+    # prefill is populated with decoding tokens and we test that it
+    # doesn't fail This test might fail if cache is not allocated
+    # correctly for n > 1 decoding steps inside a
+    # chunked prefill forward pass (where we have both prefills
+    # and decoding together )
+    sampling_params = SamplingParams(n=3,
+                                     temperature=1,
+                                     seed=0,
+                                     max_tokens=max_tokens)
+    with vllm_runner(
+            model,
+            dtype=dtype,
+            enable_chunked_prefill=True,
+            max_num_batched_tokens=30,
+            max_num_seqs=10  # forces prefill chunks with decoding
+    ) as vllm_model:
+        vllm_model.generate(example_prompts, sampling_params)
+
+
+@pytest.mark.parametrize("model", MODELS)
+@pytest.mark.parametrize("dtype", ["float"])
+@pytest.mark.parametrize("max_tokens", [96])
+def test_mamba_prefill_chunking(hf_runner, vllm_runner, example_prompts,
+                                model: str, dtype: str,
+                                max_tokens: int) -> None:
+
+    with vllm_runner(model, dtype=dtype,
+                     enable_chunked_prefill=False) as vllm_model:
+        non_chunked = vllm_model.generate_greedy([example_prompts[0]],
+                                                 max_tokens=max_tokens)
+
+    with vllm_runner(
+            model,
+            dtype=dtype,
+            enable_chunked_prefill=True,
+            max_num_batched_tokens=5,  # doesn't allow prompt longer than 10
+            max_num_seqs=3  # forces prefill chunks with decoding
+    ) as vllm_model:
+        chunked = vllm_model.generate_greedy([example_prompts[0]],
```

Comment:
Hi @mzusman  ! 
Are prefills being scheduled with decodes here ?  
looking at  `[example_prompts[0]]`, it looks like only one sequence is being submitted ? Maybe I am missing something ?

Reply:
Hi!
I've fixed this test, I was doing this since some of the example prompts had a numerical error that changed the generation therefore it failed, I've removed those problematic sequences from this test and now it's ok